features/0685-pickup-v2/README.md

```diff
@@ -0,0 +1,184 @@
+# 0212: Pickup Protocol 2.0
+
+- Authors: [Sam Curren](telegramsam@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2019-09-03
+- Status Note: Initial version
+- Start Date: 2020-12-22
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+
+## Summary
+
+A protocol to facilitate an agent picking up messages held at a mediator.
+
+## Motivation
+
+Messages can be picked up simply by sending a message to the _Mediator_ with a `return_route` decorator specified. This mechanism is implicit, and lacks some desired behavior made possible by more explicit messages.
+
+This protocol is the explicit companion to the implicit method of picking up messages.
+
+## Tutorial
+
+### Roles
+
+**Mediator** - The agent that has messages waiting for pickup by the _recipient_.
+**Recipient** - The agent who is picking up messages.
+
+### Flow
+
+status can be used to see how many messages are pending. The Recipient sends a StatusRequest message to the Mediator.
+
+Messages are retrieved when the Recipient  sends a `delivery_request` message to the Mediator.
+
+Message delivery is confirmed with a `message_received` acknowledgement, at which point the message can be removed from the mediator's queue.
+
+When live delivery is enabled, messages that arrive when an existing connection exists are delivered over the connection immediately.
+
+
+
+## Reference
+
+Each message sent should use the ~transport decorator as follows. This has been ommitted from the examples for brevity.
+
+```json=
+"~transport": {
+    "return_route": "all"
+}
+```
+
+
+### StatusRequest
+
+Sent by the _recipient_ to the _mediator_ to request a `status` message.
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/status-request",
+    "recipient_key": "<key for messages>"
+}
+```
+
+- `recipient_key` is optional. When specified only return status related to that recipient key. This allows the _Recipient_ to discover if any messages are in the queue that were sent to a specific key.
+
+### Status
+
+Status details about pending messages
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/status",
+    "recipient_key": "<key for messages>",
+    "message_count": 7,
+    "duration_waited": 3600,
+    "newest_time": "2019-05-01 12:00:00Z",
```

Comment:
Is this the time the mediator received the message? What if the mediator who's holding the message in turn used a pickup protocol to get the messages, and *knows* that the messages it's holding were actually held upstream for a week before it ever received them?